src/Processors/Processor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Processors;
+
+public abstract class Processor : BaseThread
+{
+    private bool _isPaused = false;
+    private bool _onPauseOnce = false;
+    private bool _onResumeOnce = false;

```

Comment:
Do you really need this variable?

Reply:
I'm responding it in the next conversation.

Reply:
I reworked this class. Please check the updated version.